packages/zosjobs/src/SubmitJobs.ts

```diff
@@ -248,14 +247,30 @@ export class SubmitJobs {
             });
             return activeJob;
         }
-        // if viewAppSpoolContent option passed, it waits till job status is output
-        // then get content of each spool file and return array of ISpoolFiles object
-        if (parms.viewAllSpoolContent || parms.waitForOutput) {
+        // wait for job status to be OUTPUT for all other situations
+        if (parms.task != null) {
+            parms.task.statusMessage = "Waiting for " + responseJobInfo.jobid + " to enter OUTPUT";
+            parms.task.percentComplete = TaskProgress.THIRTY_PERCENT;
+        }
+        const job: IJob = await MonitorJobs.waitForJobOutputStatus(session, responseJobInfo);
+        if (parms.directory) {
+            const downloadParms: IDownloadAllSpoolContentParms = {
+                jobid: job.jobid,
+                jobname: job.jobname,
+                outDir: parms.directory
+            };
+            if (parms.extension) {
+                downloadParms.extension = IO.normalizeExtension(parms.extension);
+            }
             if (parms.task != null) {
-                parms.task.statusMessage = "Waiting for " + responseJobInfo.jobid + " to enter OUTPUT";
-                parms.task.percentComplete = TaskProgress.THIRTY_PERCENT;
+                parms.task.statusMessage = "Downloading spool content for " + job.jobid +
+                    (job.retcode == null ? "" : ", " + job.retcode);
+                parms.task.percentComplete = TaskProgress.SEVENTY_PERCENT;
             }
-            const job: IJob = await MonitorJobs.waitForJobOutputStatus(session, responseJobInfo);
+            await DownloadJobs.downloadAllSpoolContentCommon(session, downloadParms);
+            return job;
+        }
+        if (parms.viewAllSpoolContent || parms.waitForOutput) {
```

Comment:
Should we use `else if` here? The original code had:
```
if (parms.viewAllSpoolContent || parms.waitForOutput) {
...
} else if (parms.directory) {
```

Flipping the order of these conditions makes sense to fix the linked issue 👍 

I believe we need to keep the `else if` like the following:
```
if (parms.directory) {
...
} else if (parms.viewAllSpoolContent || parms.waitForOutput) {
```

Otherwise we may end up fetching all spool files twice when both `--directory` and `--wait-for-output` are specified.